### DISCUSSION TITLE
Delivering static assets from source directory

### DISCUSSION THREAD

**Discussion by magoerlich** [vuejs/vitepress#3708] [Q&A] [answered]
Hello everyone,

is there any way to serve files that are within the source dir as static assets?

I have a pre-existing git repo containing documentation of our work projects in MD files.
But the Repo also contains some common script files (ps1 and sql) for reference.

My goal would be to just dump that repo into the src directory so the MD files are rendered as html and any links to the scriptfiles would enable the users to download the according files.

It works so far that the MD files are rendered as expected.
But unfortunatly in the build process the script files are sorted out/not copied over.
Of course you might think. Becaus they need to be in the public dir.

But that would mean that i have to copy all of them over there on each build, either by hand or automatically.
And to keep the links in the MD docs intact I'd have to copy the whole dir structure over there, which i would like to avoid.

Another option I see would be to just copy the files without structure and changing all the links in my Markdown documents, but i don't like that solution even less because I'd like to avoid changing the original repo as less as possible.

So is there any way to tell vitepress "Anything in the src dir thats not an .md, is a static asset"?
I tried to find a solution to this by reading the docs and googlefu, but couldn't find anyone with a similar use case.

Thank you very much.

**Comment by magoerlich** [comment]
Has anyone some idea?

**Reply by brc-dd** [reply, marked as answer]
Copying with correct directory structure is the only way IMO:

```ts
// .vitepress/config.ts

import fg from 'fast-glob'
import fs from 'node:fs/promises'
import { dirname } from 'node:path'
import { defineConfig } from 'vitepress'

export default defineConfig({
  // ...

  async buildEnd({ srcDir: src, outDir: dest }) {
    const files = await fg(['**/*', '!**/*.md'], { cwd: src, absolute: true })
    await Promise.all(
      files.map(async (file) => {
        const destFile = file.replace(src, dest)
        await fs.mkdir(dirname(destFile), { recursive: true })
        await fs.copyFile(file, destFile)
      })
    )
  }
})
```

**Reply by magoerlich** [reply]
Thanks for clarifying. Just wanted to make sure there is no built-in solution for this before implementing a custom one.
